cognite/neat/utils/cdf_loaders/_data_modeling.py

```diff
@@ -160,22 +163,26 @@ def _retrieve_view_ancestors(self, parents: list[ViewId], cache: dict[ViewId, Vi
             parents: The parents of the view to retrieve all ancestors for
             cache: The cache to store the views in
         """
-        parent_ids = parents
+        parent_ids = parents.copy()
         found: list[View] = []
+        found_ids: set[ViewId] = set()
         while parent_ids:
-            to_lookup = []
+            to_lookup: set[ViewId] = set()
             grand_parent_ids = []
             for parent in parent_ids:
-                if parent in cache:
+                if parent in found_ids:
+                    continue
```

Comment:
The current implementation can return the same view multiple times, this fixes that.